### PR TITLE
docs: how to build and test locally

### PR DESCRIPTION
**Pull request by michaelhaaf**
Figured since I'm going through the motions of setting up local testing, I might as well document how I'm doing it.

### CODE REVIEW

**Comment by xynydev** [comment]
Thanks! I've been intentionally holding off on adding this page, as the commands used to generate the Containerfile and switch to the local build will be refactored soon. 

The installation instructions should be offloaded to the CLI README. Appreciate the work on that front too, feel free to PR those changes in there.

**Comment by michaelhaaf** [comment]
Sounds good @xynydev -- I've separated out the installation and distrobox tips into a PR for the cli readme as you've suggested (see [here](https://github.com/blue-build/cli/pull/146)).

I did happen to see https://github.com/blue-build/cli/issues/65 and https://github.com/blue-build/website/issues/22 after I got started on this, and I can appreciate waiting until the interface is finalized before pushing this.

What I have done in the time since you last looked is marked certain sections that will be affected by the refactor with ```caution``` blocks. 

I want to suggest that this doc could be incorporated even before the interface change? Though I'm not certain how much you intend to change. My motivation for doing this documentation change was fundamentally about trying to understand why my github action builds were failing. 

Regardless of the upcoming interface change, I have found it valuable to be able to reliabily (i.e. without forgetting how to set things up) reproduce local builds; if my github action builds continue to fail, I could point to these docs to show what I've done to experiment. If that makes sense?

**Comment by xynydev** [comment]
This is really great! I agree about merging even before the refactor. I'll give some comments on this right now.

File: src/content/docs/how-to/local.mdx [hunk 1]
```diff
@@ -1,12 +1,171 @@
...
+[refactor-discussion]: https://github.com/blue-build/cli/issues/65
+
+If you want to do any of the following things locally, without depending on the [BlueBuild GitHub Action][bluebuild-cli-ghaction]:
+
+- preview the `Containerfile` generated your `recipe.yml` locally;
+- test your `recipe.yml` build process locally before pushing to GitHub;
+- install a locally built image onto a virtual machine;
+- rebase your currently running operating system onto a locally built image;
+
+This guide shows you how to use [Bluebuild CLI][bluebuild-cli-repo] to do so.
```
Comment (xynydev) [inline src/content/docs/how-to/local.mdx:24]:
```suggestion
This guide shows you how to use [Bluebuild CLI][bluebuild-cli-repo] to do the following things locally, without depending on the [BlueBuild GitHub Action][bluebuild-cli-ghaction]:

- preview the `Containerfile` generated your `recipe.yml` locally;
- test your `recipe.yml` build process locally before pushing to GitHub;
- install a locally built image onto a virtual machine;
- rebase your currently running operating system onto a locally built image;
```

File: src/content/docs/how-to/local.mdx [hunk 2]
```diff
@@ -1,12 +1,171 @@
...
+
+## Installing BlueBuild CLI
+
+There are several methods available for installing the BlueBuild CLI tool, documented in the the [BlueBuild CLI README][bluebuild-cli-readme].
```
Comment (xynydev) [inline src/content/docs/how-to/local.mdx:28]:
```suggestion
The BlueBuild CLI tool is installed by default on images built using BlueBuild. There are also several methods available for installing the BlueBuild CLI tool, documented in the the [BlueBuild CLI README][bluebuild-cli-readme].
```

File: src/content/docs/how-to/local.mdx [hunk 3]
```diff
@@ -1,12 +1,171 @@
...
+  bug-report   Create a pre-populated GitHub issue with information about your configuration
+  completions  Generate shell completions for your shell to stdout
+  help         Print this message or the help of the given subcommand(s)
+
+Options:
+  -v, --verbose...  Increase logging verbosity
+  -q, --quiet...    Decrease logging verbosity
+  -h, --help        Print help
+  -V, --version     Print version
+```
```
Comment (xynydev) [inline src/content/docs/how-to/local.mdx:56]: I'm not sure about statically including the whole output of the help command on here, as that is subject to updates, and is available for all local users.

File: src/content/docs/how-to/local.mdx [hunk 4]
```diff
@@ -1,12 +1,171 @@
...
+
+Example use cases for some of the commands follow in the sections below.
+
+### Completions
```
Comment (xynydev) [inline src/content/docs/how-to/local.mdx:60]: The tabs here look really good, and I wish those were used on other pages too, lol. But, I think this is also something that is mostly irrelevant to this guide and should be just featured in the CLI README, as a kind of secondary installation step.

File: src/content/docs/how-to/local.mdx [hunk 5]
```diff
@@ -1,12 +1,171 @@
...
+
+### Bug-Report: Getting Help
+
+You can autogenerate a GitHub Issue with OS/environment/version metadata using the `bug-report` command:
+
+```bash
+$ bluebuild bug-report
+```
+
+The command will ask for recipe information and will present a confirmation dialog before generating a draft issue. You can see an example of an issue generated using this command [here][sample-bug-report].
```
Comment (xynydev) [inline src/content/docs/how-to/local.mdx:171]: This is also something mostly irrelevant to this page. I've been meaning to add this to the [troubleshooting page](https://blue-build.org/learn/troubleshooting/), which I've been slowly filling in at recently.

Comment (michaelhaaf) [reply src/content/docs/how-to/local.mdx]: Fair enough, removed this in latest commit

File: src/content/docs/how-to/local.mdx (hunk 4 above)
Comment (michaelhaaf) [reply src/content/docs/how-to/local.mdx]: It is a cool component! I use them all the time in other projects. But yeah, fair enough, I've added them to the CLI instead (see https://github.com/blue-build/cli/pull/149/)

File: src/content/docs/how-to/local.mdx (hunk 3 above)
Comment (michaelhaaf) [reply src/content/docs/how-to/local.mdx]: Agree

**Comment by xynydev** [comment]
Great work! Still planning to add the missing sections?

**Comment by michaelhaaf** [comment]
@xynydev `build` is the only command I really used, this is all I was planning to write at this time

**Comment by xynydev** [comment]
Aight, I'm going to merge this then, and fill in later with the rest of the details.

**Review by xynydev — APPROVED** [review]

**Pull request merged**